From d706c4670cac0bc692b03416180e74c3380d81b0 Mon Sep 17 00:00:00 2001
From: David McKay <david.mckay@st.com>
Date: Sun, 8 Feb 2009 21:29:36 +0000
Subject: [PATCH] sh: set auto PCI cache line size

The auto PCI setup code does not set a value for the PCI cache line size
register. This can make a big difference to performance, as it will
allow the card to use more optimal PCI cycles.

Signed-off-by: David McKay <david.mckay@st.com>
---
 arch/sh/drivers/pci/pci-auto.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/arch/sh/drivers/pci/pci-auto.c b/arch/sh/drivers/pci/pci-auto.c
index 224e007..57f2438 100644
--- a/arch/sh/drivers/pci/pci-auto.c
+++ b/arch/sh/drivers/pci/pci-auto.c
@@ -521,6 +521,11 @@ pciauto_bus_scan(struct pci_channel *hose, int top_bus, int current_bus)
 					PCI_LATENCY_TIMER, 0x80);
 #endif
 
+		/* We should set a sensible value for cache line size, it makes a big difference to
+		 * performance potentially
+		 */
+		early_write_config_byte(hose, top_bus, current_bus, pci_devfn, PCI_CACHE_LINE_SIZE,L1_CACHE_BYTES >> 2);
+
 		/* Allocate PCI I/O and/or memory space */
 		pciauto_setup_bars(hose, top_bus, current_bus, pci_devfn, PCI_BASE_ADDRESS_5);
 	}
-- 
1.5.6.3

